### PR TITLE
Added cacheViewer to MacOS test app.

### PR DESCRIPTION
**Pull request by rohitnarula7176**

### CODE REVIEW

File: MSAL/MSAL.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -4395,6 +4399,7 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = D65A6FEB1E40002800C69FBA /* msal__framework__mac.xcconfig */;
 			buildSettings = {
+				CODE_SIGN_IDENTITY = "-";
```
Comment (jasoncoolmax) [inline MSAL/MSAL.xcodeproj/project.pbxproj]: Just curious, what does it mean by adding a dash here? :)

File: MSAL/test/app/mac/MSALCacheViewController.m [hunk 2]
```diff
@@ -26,16 +26,305 @@
...
+    {
+        [self deleteAppMetadata:(MSIDAppMetadataCacheItem *)item];
+    }
+}
+
+#pragma mark NSOutlineView Data Source Test Methods
+
+- (NSInteger)outlineView:(NSOutlineView *)outlineView numberOfChildrenOfItem:(id)item
+{
+    return !item ? [[self.cacheDict allKeys] count] : [[self.cacheDict objectForKey:item] count];
```
Comment (jasoncoolmax) [inline MSAL/test/app/mac/MSALCacheViewController.m]: nit: will it be more natural if we do it like `item? [[self.cacheDict objectForKey:item] count] : [[self.cacheDict allKeys] count]` :)

File: MSAL/test/app/mac/MSALCacheViewController.m [hunk 3]
```diff
@@ -26,16 +26,305 @@
...
+    {
+        NSString *textValue;
+        NSTableCellView *cellView = [outlineView makeViewWithIdentifier:@"cacheCell" owner:self];
+        
+        id parent = [outlineView parentForItem:item];
+        if (!parent)
+        {
+            textValue = [self getUPN:item] ? [self getUPN:item] : s_appMetadata;
+        }
+
```
Comment (jasoncoolmax) [inline MSAL/test/app/mac/MSALCacheViewController.m]: shall we remove this line for consistent code style? :)

File: MSAL/test/app/mac/MSALCacheViewController.m [hunk 4]
```diff
@@ -26,16 +26,305 @@
...
+#pragma mark NSOutlineView Data Source Test Methods
+
+- (NSInteger)outlineView:(NSOutlineView *)outlineView numberOfChildrenOfItem:(id)item
+{
+    return !item ? [[self.cacheDict allKeys] count] : [[self.cacheDict objectForKey:item] count];
+}
+
+- (BOOL)outlineView:(NSOutlineView *)outlineView isItemExpandable:(id)item
+{
+    return !item ? YES : [[self.cacheDict objectForKey:item] count];
```
Comment (jasoncoolmax) [inline MSAL/test/app/mac/MSALCacheViewController.m]: same comment as above :)

File: MSAL/test/app/mac/MSALCacheViewController.m [hunk 5]
```diff
@@ -26,16 +26,305 @@
...
+}
+
+- (id)outlineView:(NSOutlineView *)outlineView child:(NSInteger)index ofItem:(id)item
+{
+    return !item ? [[self.cacheDict allKeys] objectAtIndex:index] : [[self.cacheDict objectForKey:item] objectAtIndex:index];
```
Comment (jasoncoolmax) [inline MSAL/test/app/mac/MSALCacheViewController.m]: same comment as above :)

File: MSAL/MSAL.xcodeproj/project.pbxproj (hunk 1 above)
Comment (rohitnarula7176) [reply MSAL/MSAL.xcodeproj/project.pbxproj]: This means Ad Hoc signing.:)

File: MSAL/test/app/mac/MSALCacheViewController.m (hunk 2 above)
Comment (rohitnarula7176) [reply MSAL/test/app/mac/MSALCacheViewController.m]: Addressed.

File: MSAL/test/app/mac/MSALCacheViewController.m (hunk 3 above)
Comment (rohitnarula7176) [reply MSAL/test/app/mac/MSALCacheViewController.m]: Addressed.

**Pull request merged**